packages/redwood/src/index.ts

```diff
@@ -0,0 +1,143 @@
+    entrypointFsDirname,
+    undefined,
+    config,
+    meta
+  );
+
+  const spawnOpts = getSpawnOptions(meta, nodeVersion);
+  await runNpmInstall(
+    entrypointFsDirname,
+    ['--prefer-offline'],
+    spawnOpts,
+    meta
+  );
+
+  debug('Running build script...');
+  await spawnAsync('yarn', ['rw', 'build'], {
+    ...spawnOpts,
+    cwd: workPath,
+    prettyCommand: 'yarn rw build',
+  });
+
+  const {
+    buildCommand = 'yarn rw db up --no-db-client --auto-approve && yarn rw build',
+  } = config;
+
+  await execCommand(buildCommand, {
+    ...spawnOpts,
+    cwd: workPath,
+  });
+
+  const apiDistPath = join(workPath, 'api', 'dist', 'functions');
+  const webDistPath = join(workPath, 'web', 'dist');
+  const lambdaOutputs: { [filePath: string]: Lambda } = {};
+  const staticOutputs = await glob('**', webDistPath);
+
+  // Each file in the `functions` dir will become a lambda
+  const functionFiles = await glob('*.js', apiDistPath);
+
+  for (const [funcName, fileFsRef] of Object.entries(functionFiles)) {
+    const outputName = join('api', parsePath(funcName).name); // remove `.js` extension
+    const absEntrypoint = fileFsRef.fsPath;
+    const dependencies: string[] = await getDependencies(
+      absEntrypoint,
+      workPath
+    );
+    const relativeEntrypoint = relative(workPath, absEntrypoint);
+    const awsLambdaHandler = getAWSLambdaHandler(relativeEntrypoint, 'handler');
+
+    const lambdaFiles: Files = {
+      [`${LAUNCHER_FILENAME}.js`]: new FileBlob({
+        data: makeAwsLauncher({
+          entrypointPath: `./${relativeEntrypoint}`,
+          bridgePath: `./${BRIDGE_FILENAME}`,
+          helpersPath: `./${HELPERS_FILENAME}`,
+          sourcemapSupportPath: `./${SOURCEMAP_SUPPORT_FILENAME}`,
+          shouldAddHelpers: false,
+          shouldAddSourcemapSupport: false,
+          awsLambdaHandler,
+        }),
+      }),
+      [`${BRIDGE_FILENAME}.js`]: new FileFsRef({
+        fsPath: join(__dirname, 'bridge.js'),
+      }),
+    };
+
+    dependencies.forEach(fsPath => {
+      lambdaFiles[relative(workPath, fsPath)] = new FileFsRef({ fsPath });
+    });
+
+    lambdaFiles[relative(workPath, fileFsRef.fsPath)] = fileFsRef;
+
+    const lambda = await createLambda({
+      files: lambdaFiles,
+      handler: `${LAUNCHER_FILENAME}.launcher`,
+      runtime: nodeVersion.runtime,
+      environment: {},
+    });
+    lambdaOutputs[outputName] = lambda;
+  }
+
+  return {
+    output: { ...staticOutputs, ...lambdaOutputs },
+    routes: [{ handle: 'filesystem' }, { src: '/.*', dest: '/index.html' }],
+    watch: [],
+  };
+}
+
+function getAWSLambdaHandler(filePath: string, handlerName: string) {
+  const { dir, name } = parsePath(filePath);
+  return `${dir}${dir ? sep : ''}${name}.${handlerName}`;
+}
+
+export async function prepareCache({
+  workPath,
+}: PrepareCacheOptions): Promise<Files> {
+  const cache = await glob('node_modules/**', workPath);
+  return cache;
+}
+
+export { shouldServe };
```

Comment:
I don't think the default `shouldServe()` will work in this case, as it's only really applicable to `version = 3` Runtimes. Did you test this with `vc dev`?

Reply:
I haven't thought through `vc dev` yet because the redwood dev server provides serverless functions. I guess I'll have to add proxy support here to make sure everything goes through to the redwood dev server

Reply:
Ok I removed `shouldServe` and returned `routes` and `watch` when `meta.isDev` is true